wandb/sdk/wandb_settings.py

```diff
@@ -194,17 +178,65 @@ class Settings(six.with_metaclass(CantTouchThis, object)):
 
     """
 
-    # pylint: disable=no-member
-    entity: Optional[str]
+    mode: Optional[str] = None
```

Comment:
As far as I understand these are optional as arguments, but they'll always exist on the object because they get populated with defaults, right? It'd be cool if there were some way to annotate that -- in TypeScript we could have one be a `Partial` of the other -- but I'm not sure we have the ability to do that in Python.

Reply:
yep, its all evolving.  type annotation does't start helping until we get more things typed.. Figured this was a good place.   I will remove Optional tags when possible